lib/compat/wordpress-6.0/theme.json

```diff
@@ -185,6 +185,85 @@
 			],
 			"text": true
 		},
+		"layout": {
+			"definitions": {
+				"default": {
+					"name": "default",
+					"slug": "flow",
+					"className": "is-layout-flow",
+					"baseStyles": [
+						{
+							"selector": " > .alignleft",
+							"rules": {
+								"float": "left",
+								"margin-inline-start": "0",
+								"margin-inline-end": "2em"
+							}
+						},
+						{
+							"selector": " > .alignright",
+							"rules": {
+								"float": "right",
+								"margin-inline-start": "2em",
+								"margin-inline-end": "0"
+							}
+						},
+						{
+							"selector": " > .aligncenter",
+							"rules": {
+								"margin-left": "auto !important",
+								"margin-right": "auto !important"
+							}
+						}
+					],
+					"blockGapStyles": [
+						{
+							"selector": " > *",
+							"rules": {
+								"margin-block-start": "0",
+								"margin-block-end": "0"
+							}
+						},
+						{
+							"selector": " > * + *",
+							"rules": {
+								"margin-block-start": null,
+								"margin-block-end": "0"
+							}
+						}
+					]
+				},
+				"flex": {
+					"name": "flex",
+					"slug": "flex",
+					"className": "is-layout-flex",
+					"baseStyles": [
+						{
+							"selector": "",
+							"rules": {
+								"display": "flex",
+								"flex-wrap": "wrap",
+								"align-items": "center"
+							}
+						},
+						{
+							"selector": " > *",
+							"rules": {
+								"margin": "0"
+							}
+						}
+					],
+					"blockGapStyles": [
+						{
+							"selector": "",
+							"rules": {
+								"gap": null
+							}
+						}
+					]
+				}
+			}
+		},
```

Comment:
Yeah, totally — I imagined it's not something we'd advertise per se, but it could be useful in the future? If we want to ensure that themes can't override it, we could always grab the layout definitions just from the core theme.json data, rather than the merged one (possibly via `get_core_data` on the resolver class 🤔)